ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/thread/ThreadActionData.java

```diff
@@ -242,7 +242,7 @@ public static void purgeActions (Thread t, boolean unregister)
 
 					if (act != null)
                     {
-						act.removeChildThread(ThreadUtil.getThreadId(t));
+						act.removeChildThread(ThreadUtil.removeThreadId(t));
                     }
```

Comment:
Can it return null at this point of the execution though? From what I can see it should not be possible.